tf_quant_finance/black_scholes/approximations/american_option_test.py

```diff
@@ -338,6 +338,139 @@ def test_bs1993_prices_long_term2(self, discount_rates, dividend_rates,
       self.assertAllClose(expected_prices, computed_prices,
                           rtol=5e-3, atol=5e-3,
                           msg=msg)
+  
+  @parameterized.parameters(
+      (0.08, 0.12, 0.2, 0.25,
```

Comment:
could you please add a test where all inputs are ternsors and where some options are puts and some are calls?

Reply:
Added.  Let me know if this isn't quite what you were looking for.

Reply:
Sorry, I did not explain myself well. I meant, at the moment rates and expiries are scalars. Could you make them tensors of the same shape as spots. E.g., for discount rates `discount_rates` use `2 * [0.08, 0.07, 0.06, 0.05, 0.04]`. Just one test will do. This way we know that batching works.